libautoscoper/src/Tracker.cpp

```diff
@@ -419,61 +419,41 @@ void Tracker::optimize(int frame, int dFrame, int repeats, int opt_method, unsig
       unsigned int MAX_EPOCHS = max_iter;
       unsigned int MAX_STALL = max_stall_iter;
 
-      float positions[NUM_OF_PARTICLES*NUM_OF_DIMENSIONS];
-      float velocities[NUM_OF_PARTICLES*NUM_OF_DIMENSIONS];
-      float pBests[NUM_OF_PARTICLES*NUM_OF_DIMENSIONS];
-      float gBest[NUM_OF_DIMENSIONS];
+      std::vector<Particle*> particles;
+      Particle* gBest = new Particle();
```

Comment:
```suggestion
     Particle gBest;
```

I suggest to manage the object on stack. Since it has only one float, and two small vectors, it should now impact and will simplify the code.